TF_main_loop/validate.py

```diff
@@ -170,24 +227,42 @@ def save_images(this_set, x_batch, y_batch, f_batch, y_pred_batch,
         else:
             of = None
 
+        r_data = x
+
+        if x.shape[-1] == 5:
+            r_data = raw_data[seq_length // 2, ..., :3]
+
         # PRINT THE HEATMAP
         if save_heatmap:
             # do not pass optical flow
-            save_heatmap_fn(raw_data[seq_length // 2, ..., :3], of,
-                            y_pred, labels, save_basedir, subset, f)
+            save_heatmap_fn(r_data, of, y_soft_pred,
+                            labels, nclasses,
+                            save_basedir, subset,
+                            f, img_summaries,
+                            epoch_id, save_images_on_disk,
+                            summary_writer)
 
         # PRINT THE SAMPLES
         # Keep most likely prediction only
-        y = y.argmax(2)
-        y_pred = y_pred.argmax(2)
+        # y = y.argmax(2)
+        # y_pred = y_pred.argmax(2)
+
+        r_data = x
```

Comment:
I don't think this is right, you want `raw_data` not `x`, cause `raw_data` will be the original image while `x` will be the image after the transformations (e.g., after normalization).

I suggest you remove this, make L253 `raw_data = raw_data[seq_length // 2]` and pass `raw_data` to  `save_sample_and_fill_sequence_fn` at L258